File: frontend/lib/features/home/data/dive_site_model.dart
class DiveSiteModel {
  final String id;
  final String nome;
  final String? descricao;
  final String pais;
  final String? estado;
  final String? cidade;
  final double latitude;
  final double longitude;
  final double? profundidadeMaxima;
  final String? tipo;

  const DiveSiteModel({required this.id, required this.nome, this.descricao, required this.pais, this.estado, this.cidade, required this.latitude, required this.longitude, this.profundidadeMaxima, this.tipo});

  factory DiveSiteModel.fromJson(Map<String, dynamic> json) => DiveSiteModel(
    id: json['id'].toString(), nome: json['nome'] ?? '', descricao: json['descricao'], pais: json['pais'] ?? '', estado: json['estado'], cidade: json['cidade'], latitude: (json['latitude'] as num).toDouble(), longitude: (json['longitude'] as num).toDouble(), profundidadeMaxima: (json['profundidadeMaxima'] as num?)?.toDouble(), tipo: json['tipo'],
  );
}

File: backend/src/main/java/com/log/diverr/diver/divesites/domain/DiveSiteModel.java
package com.log.diverr.diver.divesites.domain;

import com.log.diverr.diver.users.domain.UserModel;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@Setter
@Entity
@Table(name = "dive_sites")
public class DiveSiteModel {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String nome;

    @Column(length = 1000)
    private String descricao;

    @Column(nullable = false)
    private String pais;

    private String estado;

    private String cidade;

    @Column(nullable = false, precision = 10, scale = 7)
    private BigDecimal latitude;

    @Column(nullable = false, precision = 10, scale = 7)
    private BigDecimal longitude;

    private Double profundidadeMaxima;

    @Enumerated(EnumType.STRING)
    private TipoDiveSite tipo;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by")
    private UserModel createdBy;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    public DiveSiteModel() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
    }

    public enum TipoDiveSite {
        COSTEIRO,
        NAUFRAGIO,
        RECIFE,
        CAVERNA,
        PAREDE,
        PEDREIRA,
        LAGO,
        RIO,
        OUTRO
    }
}